scrapbook/models.py

```diff
@@ -0,0 +1,312 @@
+        return scraps
+
+    @property
+    def scraps(self):
+        """dict: a dictionary of data found in the notebook"""
+        if self._scraps is None:
+            self._scraps = self._fetch_scraps()
+        return self._scraps
+
+    @property
+    def cell_timing(self):
+        """list: a list of cell execution timings in cell order"""
+        return [
+            # TODO: Other timing conventions?
+            cell.metadata.get('papermill', {}).get('duration', 0.0)
+                if cell.get("execution_count") else None
+            for cell in self.node.cells
+        ]
+
+    @property
+    def execution_counts(self):
+        """list: a list of cell execution counts in cell order"""
+        return [
+            cell.get("execution_count") for cell in self.node.cells
+        ]
+
+    @property
+    def papermill_metrics(self):
+        """pandas dataframe: dataframe of cell execution counts and times"""
+        df = pd.DataFrame(columns=['filename', 'cell', 'value', 'type'])
+
+        for i, cell in enumerate(self.node.cells):
+            execution_count = cell.get("execution_count")
+            if not execution_count:
+                continue
+            name = "Out [{}]".format(str(execution_count))
+            value = cell.metadata.get('papermill', {}).get('duration', 0.0)
+            df.loc[i] = self.filename, name, value, "time (s)"
+        return df
+
+    @property
+    def parameter_dataframe(self):
+        """pandas dataframe: dataframe of notebook parameters"""
+        # Meant for backwards compatibility to papermill's dataframe method
+        return pd.DataFrame(
+            [[name, self.parameters[name], 'parameter', self.filename]
+                for name in sorted(self.parameters.keys())],
+            columns=['name', 'value', 'type', 'filename'])
+
+    @property
+    def scrap_dataframe(self):
+        """pandas dataframe: dataframe of cell scraps"""
+        # Meant for backwards compatibility to papermill's dataframe method
+        return pd.DataFrame(
+            [[name, self.scraps[name], 'record', self.filename]
+                for name in sorted(self.scraps.keys())],
+            columns=['name', 'value', 'type', 'filename'])
+
+    @property
+    def papermill_dataframe(self):
+        """pandas dataframe: dataframe of notebook parameters and cell scraps"""
+        # Meant for backwards compatibility to papermill's dataframe method
+        return self.parameter_dataframe.append(self.scrap_dataframe, ignore_index=True)
+
+    def _fetch_frames(self):
+        outputs = collections.OrderedDict()
+        for cell in self.node.cells:
+            for output in cell.get('outputs', []):
+                if 'scrapbook' in output.get('metadata', {}):
+                    output_name = output.metadata.scrapbook.get('name')
+                    if output_name:
+                        outputs[output_name] = output
+                # Backwards compatibility
```

Comment:
For backwards compatibility items, let's add a space before and after the code block that deals with compatibility (this lets a future dev know which lines are specific to backwards compatibility)